common/persistence/nosql/nosqlplugin/cassandra/gocql/native/client.go

```diff
@@ -18,15 +18,56 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package cassandra
+
+package native
 
 import (
+	"context"
+
+	gogocql "github.com/gocql/gocql"
+
 	"github.com/uber/cadence/common/persistence/nosql/nosqlplugin/cassandra/gocql"
 )
 
-var defaultGoCQLClient = gocql.NewClient()
+var _ gocql.Client = client{}
+
+type (
+	client struct {
+	}
+)
+
+func init() {
+	gocql.RegisterClient(client{})
+}
```

Comment:
Personally I would prefer to call the RegisterClient directly so it's more clear we are doing and won't surprise people when they import this package or import a package that import this package. But no strong opinion on it.

Reply:
It's certainly a good concern. 
The reason behind this is that module variable is better to be immutable once being set. Allowing override (using Replace ) is dangerous can lead to undefined behavior. 

Because of the immutability, this client can only be set once. If not using import, we have to make sure not calling RegisterClient more than noce. 

I think we have to do it anyway. Looks like your internal repo needs to call this API for testing.  

I will change to explict calling pattern.

Reply:
okay I still keep using `init`. Because it turns out that it's not possible to call only once in the test(which started by `go test ...` command). So if we want to call RegisterClient directly, we must allowing the registeredClient to be immutable.

However, I don't like the idea of adding "_ import ..." in every "..._test.go" files. Because it's error prone and people may forget about doing that. 

So I ended up making an special package "external" as a wrapper for NewTestBaseWithCassandra. This is the best idea I have right now for all the concerns.